### PR TITLE
Augment as many SWEET owl:Class with a skos:definition from wikidata

### PR DESCRIPTION
**Pull request by lewismc**
Hi @carueda 
I got some renewed motivation to work on [SWEET issue 125](https://github.com/ESIPFed/sweet/issues/125) and thought I would push my quick work to give you an idea of what my plans are. This should be a trivial script but I would appreciate your peer review as I haven't written much Scala for a few years now so i am a bit rusty syntax wise. Additionally, my local tooling isn't great so I am not catching compile errors or syntax errors just now... just trying to get as much of the pseudo code in place via my terminal.
The question I want to ask is whether we should be using [schema:description](https://schema.org/description) as is done at wikidata or else we should use some other predicate e.g. skos:... or dct:...
Do you have any comments?

### CODE REVIEW

**Review by carueda — COMMENTED** [review]
Hi Lewis,
I looked at your code and tried to do adjustments anywhere I got some sense of the intent.  As ammonite was having weird (and rather annoying) issues compiling code that was correct, I decided to put the adjusted code in a proper Scala project, which I'll be sending your way shortly (via slack) so you can see that changes and some comments. Start with the included readme.

**Comment by carueda** [comment]
Now, regarding the selection of the particular property (`schema:description`, `rdfs:comment`, etc.) or mechanism, well, very interesting [discussion](https://github.com/ESIPFed/sweet/issues/125) indeed, but sorry, I don't have anything to suggest.  Unfortunately, I haven't been able to keep up-to-date with best practices, etc.  However, happy to assist with the coding part if you'd like.

**Comment by carueda** [comment]
@lewismc  re:

> ... As ammonite was having weird (and rather annoying) issues compiling code that was correct...

I just entered https://github.com/lihaoyi/Ammonite/issues/1040

**Comment by lewismc** [comment]
Excellent @carueda I'll push some more updates and we can take it from there.

**Comment by lewismc** [comment]
Hi @carueda I pushed another commit which essentially enables the first functional implementation. There are quite a few improvements I would like to make however. If we could chat about this come Monday it would be excellent. Thanks for your assistance so far... this is looking much better already.

File: sc/augment_wikidata_definitions/build.sbt [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+organizationName := "example"
+
+lazy val root = (project in file("."))
+  .settings(
+    name := "augment_wikidata_definitions",
+
+    libraryDependencies ++= Seq(
+      ("org.apache.jena" % "jena-core" % "3.3.0").exclude("org.slf4j", "slf4j-log4j12"),
+
+      "org.apache.jena" % "jena-tdb" % "3.3.0",
```
Comment (lewismc) [inline sc/augment_wikidata_definitions/build.sbt]: Would like to update to jena 3.14.0 for starters

File: sc/augment_wikidata_definitions/src/main/scala/sweet.tools/augment_wikidata_definitions.scala [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+      }
+    }
+
+    def executeWikidataDescriptionQuery(label: String): String = {
+      val query = getWikidataDescriptionQuery(label)
+      val response = tryWith(QueryExecutionFactory.sparqlService("https://query.wikidata.org/sparql", query)){qexec =>
+        val results = qexec.execSelect()
+        if (results.hasNext()) {
+          var soln = results.next()
+          return soln.getLiteral("o").getString()
```
Comment (lewismc) [inline sc/augment_wikidata_definitions/src/main/scala/sweet.tools/augment_wikidata_definitions.scala]: I would like to change this to the `Literal` or even the `RDFNode` such that the language context is retained when we write the data back to the `Model`.

File: sc/augment_wikidata_definitions/src/main/scala/sweet.tools/augment_wikidata_definitions.scala [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+          println(s"ERROR: $file: ${e.getMessage}")
+          None
+      }
+    }
+
+    def executeWikidataDescriptionQuery(label: String): String = {
+      val query = getWikidataDescriptionQuery(label)
+      val response = tryWith(QueryExecutionFactory.sparqlService("https://query.wikidata.org/sparql", query)){qexec =>
+        val results = qexec.execSelect()
+        if (results.hasNext()) {
```
Comment (lewismc) [inline sc/augment_wikidata_definitions/src/main/scala/sweet.tools/augment_wikidata_definitions.scala]: For some reason I was unable to add multiple definitions to an `Array[String]`... I spent quite a bit of time on it but it just never seemed to work. I think it has to do with the immutability of Scala Collections so maybe we can discuss that!

File: sc/augment_wikidata_definitions/src/main/scala/sweet.tools/augment_wikidata_definitions.scala [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+    }
+
+    def getWikidataDescriptionQuery(label: String): String = {
+      s"""PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+         |PREFIX schema: <http://schema.org/>
+         |SELECT ?o WHERE {
+         |    ?s rdfs:label "${label}"@en .
+         |    ?s schema:description ?o .
+         |    FILTER(LANGMATCHES(LANG(?o), "en"))
+         |    FILTER(STRLEN(?o) > 15)
```
Comment (lewismc) [inline sc/augment_wikidata_definitions/src/main/scala/sweet.tools/augment_wikidata_definitions.scala:149]: This `FILTER` clause is not particularly logical. I would like something better which determines whether the `schema:description` actually makes sense for the SWEET label...

File: sc/augment_wikidata_definitions/src/main/scala/sweet.tools/augment_wikidata_definitions.scala [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+object augment_wikidata_definitions {
+
+  def main(args: Array[String]): Unit = {
+    if (args.nonEmpty) {
+      main(args.head)
+    }
+    else println("Missing directory argument")
+  }
+
+  val schemaDescriptionProp = ResourceFactory.createProperty("http://schema.org/description")
```
Comment (lewismc) [inline sc/augment_wikidata_definitions/src/main/scala/sweet.tools/augment_wikidata_definitions.scala]: This ends up being added as `http://schema.org/description` for every predicate... I would rather introduce a prefix and then use the condensed version e.g. `schema:description`. do you know how to do this?

**Review by lewismc — COMMENTED** [review]
I did some self critique and have been addressing some of the issues.

**Pull request merged**